--- origsrc/libxcrypt-4.4.4/lib/crypt-port.h	2018-12-22 04:15:58.000000000 -0600
+++ src/libxcrypt-4.4.4/lib/crypt-port.h	2019-01-02 19:06:08.610351500 -0600
@@ -177,8 +177,12 @@ _crypt_strcpy_or_abort (void *, const si
 
 /* Set the symbol version for EXTNAME, which uses INTNAME as its
    implementation.  */
+#ifdef __ELF__
 #define symver_set(extstr, intname, version, mode) \
   __asm__ (".symver " #intname "," extstr mode #version)
+#else
+#define symver_set(...) symver_nop()
+#endif
 
 /* A construct with the same syntactic role as the expansion of symver_set,
    but which does nothing.  */
