--- origsrc/libxcrypt-4.4.4/test/crypt-badargs.c	2018-12-22 04:15:58.000000000 -0600
+++ src/libxcrypt-4.4.4/test/crypt-badargs.c	2019-01-02 20:49:10.556640600 -0600
@@ -79,7 +79,7 @@ static const char *settings[] =
 };
 
 /* In some of the tests below, a segmentation fault is the expected result.  */
-static jmp_buf env;
+static sigjmp_buf env;
 static void
 segv_handler (int sig)
 {
--- origsrc/libxcrypt-4.4.4/test/getrandom-interface.c	2018-12-22 04:15:58.000000000 -0600
+++ src/libxcrypt-4.4.4/test/getrandom-interface.c	2019-01-02 20:49:37.762695300 -0600
@@ -126,7 +126,7 @@ test_fault (char *page, size_t pagesize)
 }
 
 /* In one of the tests above, a segmentation fault is the expected result.  */
-static jmp_buf env;
+static sigjmp_buf env;
 static void
 segv_handler (int sig)
 {
