--- origsrc/libxcrypt-4.4.2/Makefile.am	2019-01-02 19:17:32.038085900 -0600
+++ src/libxcrypt-4.4.2/Makefile.am	2019-01-02 20:24:56.379882800 -0600
@@ -86,6 +86,8 @@ else
 
 if HAVE_VSCRIPT
 libcrypt_la_LDFLAGS += $(VSCRIPT_LDFLAGS),$(builddir)/libcrypt.map
+else
+libcrypt_la_LDFLAGS += -no-undefined -export-symbols $(srcdir)/lib/libcrypt.def
 endif
 
 libcrypt_la_LDFLAGS += $(UNDEF_FLAG) $(TEXT_RELOC_FLAG) $(AM_LDFLAGS)
--- origsrc/libxcrypt-4.4.2/lib/libcrypt.def	1969-12-31 18:00:00.000000000 -0600
+++ src/libxcrypt-4.4.2/lib/libcrypt.def	2019-01-02 20:19:29.947265600 -0600
@@ -0,0 +1,10 @@
+EXPORTS
+    crypt = _crypt_crypt
+    crypt_r = _crypt_crypt_r
+    crypt_ra = _crypt_crypt_ra
+    crypt_rn = _crypt_crypt_rn
+    crypt_gensalt = _crypt_crypt_gensalt
+    crypt_gensalt_ra = _crypt_crypt_gensalt_ra
+    crypt_gensalt_rn = _crypt_crypt_gensalt_rn
+    crypt_checksalt = _crypt_crypt_checksalt
+    crypt_preferred_method = _crypt_crypt_preferred_method
