--- origsrc/nano-2.9.7/syntax/patch.nanorc	2018-04-10 04:57:51.000000000 -0500
+++ src/nano-2.9.7/syntax/patch.nanorc	2018-05-17 10:52:47.301467600 -0500
@@ -1,6 +1,6 @@
 ## Here is an example for patch files.
 
-syntax patch "\.(patch|diff|debdiff)$"
+syntax patch "\.(patch|diff|debdiff)$" "COMMIT_EDITMSG$"
 magic "diff output"
 # There is no official support for comments in patch files.
 comment ""
