--- origsrc/nano-4.9/doc/sample.nanorc.in	2020-03-24 06:03:28.000000000 -0400
+++ src/nano-4.9/doc/sample.nanorc.in	2020-03-29 13:44:06.382460800 -0400
@@ -239,7 +239,7 @@
 ##   include "/path/to/syntax_file.nanorc"
 
 ## To include all existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 
 ## If <Tab> should always produce four spaces when editing a Python file,
