--- origsrc/nano-2.9.7/syntax/nanorc.nanorc	2018-05-10 06:18:29.000000000 -0500
+++ src/nano-2.9.7/syntax/nanorc.nanorc	2018-05-17 10:56:26.303279800 -0500
@@ -20,7 +20,30 @@ icolor green "^[[:space:]]*((un)?(bind|s
 color brightmagenta "".+"([[:space:]]|$)"
 
 # Colors
-icolor yellow "^[[:space:]]*i?color[[:space:]]*(bright)?(white|black|red|blue|green|yellow|magenta|cyan|normal)?(,(white|black|red|blue|green|yellow|magenta|cyan|normal))?\>"
+icolor brightwhite "^[[:space:]]*i?color[[:space:]]*brightwhite\>"
+icolor white "^[[:space:]]*i?color[[:space:]]*white\>"
+icolor ,white "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,white\>"
+icolor brightblack "^[[:space:]]*i?color[[:space:]]*brightblack\>"
+icolor black "^[[:space:]]*i?color[[:space:]]*black\>"
+icolor ,black "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,black\>"
+icolor brightred "^[[:space:]]*i?color[[:space:]]*brightred\\>"
+icolor red "^[[:space:]]*i?color[[:space:]]*red\>"
+icolor ,red "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,red\>"
+icolor brightblue "^[[:space:]]*i?color[[:space:]]*brightblue\>"
+icolor blue "^[[:space:]]*i?color[[:space:]]*blue\>"
+icolor ,blue "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,blue\>"
+icolor brightgreen "^[[:space:]]*i?color[[:space:]]*brightgreen\>"
+icolor green "^[[:space:]]*i?color[[:space:]]*green\>"
+icolor ,green "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,green\>"
+icolor brightyellow "^[[:space:]]*i?color[[:space:]]*brightyellow\>"
+icolor yellow "^[[:space:]]*i?color[[:space:]]*yellow\>"
+icolor ,yellow "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,yellow\>"
+icolor brightmagenta "^[[:space:]]*i?color[[:space:]]*brightmagenta\>"
+icolor magenta "^[[:space:]]*i?color[[:space:]]*magenta\>"
+icolor ,magenta "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,magenta\>"
+icolor brightcyan "^[[:space:]]*i?color[[:space:]]*brightcyan\>"
+icolor cyan "^[[:space:]]*i?color[[:space:]]*cyan\>"
+icolor ,cyan "^[[:space:]]*i?color[[:space:]]*[[:alpha:]]*,cyan\>"
 icolor magenta "^[[:space:]]*i?color\>" "\<(start|end)="
 
 # Comments
